Sources/Player/Common/Event/NotificationsHandler.swift

```diff
@@ -58,4 +60,40 @@ final class NotificationsHandler {
 			self.listener.djSessionTransitioned(to: transition)
 		}
 	}
+
+	func offliningStarted(for mediaProduct: MediaProduct) {
+		queue.async {
```

Comment:
[question] since this is "new", have you considered using structured concurrency? So it's potentially less code to migrate when (one sunny day) we eventually move to it 🤓 or maybe a separate queue would help already, so we could migrate separately if possible.